crates/explorer/src/lib.rs

```diff
@@ -181,7 +183,21 @@ fn annotate_asm(
                     })
                 })
                 .collect::<Result<Vec<_>>>()?;
-            Ok(AnnotatedFunction { instructions })
+
+            let demangled_name = match name.splitn(2, "::").nth(1) {
+                Some(name) => {
+                    let mut demangled = String::new();
+                    demangle_function_name(&mut demangled, name)
+                        .map_or_else(|_| format!("demangle-error::{}", name), |_| demangled)
+                }
+                None => name.to_string(),
+            };
```

Comment:
I think it would be better to let `demangled_name` be an `Option<String>` rather than making a clone of `name` and then have the JS choose the demangled name when available and fall back to the plain name.

Reply:
OK.